Uplift/Views/GymDetail/FitnessCenter/GymDetailFitnessCenterHoursCell.swift

```diff
@@ -2,13 +2,238 @@
+         height += CGFloat(hours.getNumHoursLines() - 1) * LayoutConstants.hoursTableViewRowHeight
+        }
+
+        return height
+    }
+
+    public static func getHeight(isDisclosed: Bool, numLines: Int) -> CGFloat {
+        var height = LayoutConstants.hoursTableViewBottomRowHeight
+        if isDisclosed {
+            height += CGFloat(numLines - 1) * LayoutConstants.hoursTableViewRowHeight
+        }
+        height += GymDetailConstraints.verticalPadding
+        height += GymDetailConstraints.titleLabelHeight
+        height += LayoutConstants.hoursTableViewTopPadding
+        height += GymDetailConstraints.verticalPadding
+        height += GymDetailConstraints.dividerViewHeight
+        return height
+    }
+
+    @objc func dropHours(sender: UITapGestureRecognizer) {
+        hoursTableView.beginUpdates()
+        var modifiedIndices: [IndexPath] = []
+        (0..<(hours.getNumHoursLines())).forEach { i in
+            modifiedIndices.append(IndexPath(row: i, section: 0))
+        }
+
+        if isDisclosed { // collapsing details
+            isDisclosed = false
+            hoursTableView.deleteRows(at: modifiedIndices, with: .fade)
+            // swiftlint:disable:next force_cast
+            (hoursTableView.headerView(forSection: 0) as! GymHoursHeaderView).downArrow.image = .none
+            // swiftlint:disable:next force_cast
+            (hoursTableView.headerView(forSection: 0) as! GymHoursHeaderView).rightArrow.image = UIImage(named: ImageNames.rightArrowSolid)
+
+            self.delegate?.didDropHours(isDropped: false) { () in
+                UIView.animate(withDuration: 0.3) {
+                    self.tableViewHeight?.constraint.deactivate()
+
+                    self.hoursTableView.snp.makeConstraints { make in
+                        self.tableViewHeight = make.height.equalTo(self.getTableViewHeight())
+                    }
+                    self.contentView.layoutIfNeeded()
+                }
+            }
+        } else { // expanding details
+            isDisclosed = true
+            hoursTableView.insertRows(at: modifiedIndices, with: .fade)
+            // swiftlint:disable:next force_cast
+            (hoursTableView.headerView(forSection: 0) as! GymHoursHeaderView).downArrow.image = UIImage(named: ImageNames.downArrowSolid)
+            // swiftlint:disable:next force_cast
+            (hoursTableView.headerView(forSection: 0) as! GymHoursHeaderView).rightArrow.image = .none
+
+            self.delegate?.didDropHours(isDropped: true) { () in
+                UIView.animate(withDuration: 0.5) {
+                    self.tableViewHeight?.constraint.deactivate()
+
+                    self.hoursTableView.snp.makeConstraints { make in
+                        self.tableViewHeight = make.height.equalTo(self.getTableViewHeight())
+                    }
+                    self.contentView.layoutIfNeeded()
+                }
+            }
+        }
+        hoursTableView.endUpdates()
```

Comment:
can we add isDisclosed = !isDisclosed or isDisclosed.toggle() at the bottom here?

Reply:
isDisclosed needs to be set before any of the updates start. I set them manually within the actual if statement because i felt like that increased readability without having to create a new variable.